conda/core/portability.py

```diff
@@ -172,12 +187,33 @@ def replace_long_shebang(mode, data):
         shebang_match = re.match(SHEBANG_REGEX, data, re.MULTILINE)
         if shebang_match:
             whole_shebang, executable, options = shebang_match.groups()
-            if len(whole_shebang) > 127:
-                executable_name = executable.decode('utf-8').split('/')[-1]
+            prefix, executable_name = executable.decode("utf-8").rsplit("/", 1)
+            if len(whole_shebang) > 127 or "\\ " in prefix:
                 new_shebang = '#!/usr/bin/env %s%s' % (executable_name, options.decode('utf-8'))
                 data = data.replace(whole_shebang, new_shebang.encode('utf-8'))
 
     else:
         # TODO: binary shebangs exist; figure this out in the future if text works well
         pass
     return data
+
+
+def generate_shebang_for_entry_point(executable):
+    shebang = f"#!{executable}\n"
+    # In principle, this shebang ^ will work as long as the path
+    # to the python executable does not contain spaces AND it's not
+    # longer than 127 characters. But if it does, we can fix it.
+    # Following method inspired by `pypa/distlib`
+    # https://github.com/pypa/distlib/blob/91aa92e64/distlib/scripts.py#L129
+    # Explanation: these lines are both valid Python and shell :)
+    # 1. Python will read it as a triple-quoted multiline string; end of story
+    # 2. The shell will see:
+    #       * '' (empty string)
+    #       * 'exec' "path/with spaces/to/python" "this file" "arguments"
+    #       * ' ''' (quoted space followed by empty string)
+    if len(shebang) > 127 or " " in shebang:
+        shebang = "#!/bin/sh\n"
+        shebang += "'''exec' " + f'"{executable}" "$0" "$@"'
+        shebang += "\n' '''\n"
```

Comment:
Do we need to consider macOS' longer max shebang compatibility here as well? https://github.com/pypa/distlib/blob/91aa92e64/distlib/scripts.py#L145-L146